logstash-core/lib/logstash/modules/settings_merger.rb

```diff
@@ -20,4 +20,25 @@ def self.merge(cli_settings, yml_settings)
     end
     merged
   end
+
+  def self.merge_cloud_settings(module_settings, logstash_settings)
+    cloud_id = logstash_settings.get("cloud.id")
+    cloud_auth = logstash_settings.get("cloud.auth")
+    if cloud_id.nil?
+      if cloud_auth.nil?
+        return # user did not specify cloud settings
+      else
+        raise ArgumentError.new("Cloud Auth without Cloud Id")
+      end
+    end
+    module_settings["var.kibana.scheme"] = "https"
```

Comment:
Can we log here as info that we are overwriting because of a presence of cloud.id?